uvicorn/config.py

```diff
@@ -31,6 +32,11 @@
 from uvicorn.middleware.proxy_headers import ProxyHeadersMiddleware
 from uvicorn.middleware.wsgi import WSGIMiddleware
 
+if TYPE_CHECKING:
+    PathLike = os.PathLike[str]
+else:
+    PathLike = os.PathLike
```

Comment:
Why do we need to write this? 🤔

Reply:
Because `os.PathLike` is not a generic at runtime in previous Python versions.

Ref.: https://github.com/python/mypy/pull/5833/files

Reply:
In baize I write this, mypy or pylance works fine. 

```
        directory: Union[str, "os.PathLike[str]"],
```

Reply:
That's better. Thanks.

Reply:
Yes, PR welcome with that.